File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;


public interface BrandService {
	public List<TbBrand> findAll();
	
	//4.0.0做法
	//public PageResult findPage(Integer pageNum,Integer pageSize);
	
	//5.1.2做法分页查询
	public List<TbBrand> findPage1(Integer pageNum,Integer pageSize);

	/**
	 * 分页查询新版本
	 * @param pageNum 当前页
	 * @param pageSize 当前页的数据量
	 * @return 分页对象包含总数
	 */
    PageInfo findPage(Integer pageNum, Integer pageSize);

    void add(TbBrand brand);

    /**
     *
     * @param id brand id
     * @return brand
     */
    TbBrand findOne(Long id);

    /**
     *
     * @param brand 需要修改的brand
     *
     */
    void update(TbBrand brand);

    /**
     *
     * @param ids 需要删除的brandId数组
     */
    void delBrand(Long[] ids);

    /**
     *条件查询
     * @param brand 条件查询条件
     * @param pageNum 当前页
     * @param pageSize 每页数据量
     * @return 查询的分页
     */
    PageInfo searchPage(TbBrand brand, Integer pageNum, Integer pageSize);

    /**
     * 查询所有的品牌[{"id": 1, "text": "联想"}, {"id": 3, "text": "三星"}]
     * @return 返回select2下拉列表格式的数据形式
     */
    List<Map> select2BrandList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    //读取列表数据到表单
    this.findAll = function () {
        return $http.get("../brand/findPage.do");
    };
    //分页
    this.findPage = function (pageNum, pageSize) {
        return $http.get("../brand/findPage.do?pageNum=" + pageNum + "&pageSize=" + pageSize);
    };
    //查询实体
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };
    //修改
    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    };
    //增加
    this.add = function (entity) {
        return $http.post("../brand/add.do", entity);
    };
    // 删除
    this.del = function (ids) {
        return $http.get("../brand/delBrand.do?ids=" + ids);
    };
    // 搜索
    this.searchPage = function (pageNum, pageSize, searchEntity) {
        return $http.post("../brand/searchPage.do?pageNum=" + pageNum + "&pageSize=" + pageSize, searchEntity);
    };
    this.select2BrandList=function () {
        return $http.get("../brand/select2BrandList.do");
    };
});